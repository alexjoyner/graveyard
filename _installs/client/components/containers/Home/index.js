import React, { Component } from 'react';
import { connect } from 'react-redux';
import {} from '../../../js/actions/index';
require('./_.sass');
class HomeContainer extends Component {
	constructor(props){
		super(props);
		this.state = {};
	}
	render() {
		return (
			<div className="row">
				<div className="col col-lg-3 hidden-lg-down">

				</div>
				<div className="col col-lg-6 col-xs-12">
					<h1>HELLO WORLD</h1>
				</div>
				<div className="col col-lg-3 hidden-lg-down">

				</div>
			</div>
		);
	}
}

function mapStateToProps() {
	return {};
}
export default connect(mapStateToProps, {})(HomeContainer);
