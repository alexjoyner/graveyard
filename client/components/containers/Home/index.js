/*
* Our containers are the glue that holds our app together.  There should be a MINIMAL
* amount of pretty components in our app.  It should essentially manage passing our data from
* our store to each of our components. It is a container for the page we are on. Containers are
* not pretty. They just hold stuff and give you stuff. You decorate the outside of containers with
* different components of stuff to make it look nice...  Paint, Stickers, magnets, lights, horns, maps,
* usb drives, nudes, etc.  I don't care what you put on your container. I'm not telling you how to live
* your life... OK? Kapeesh? GOOD.
* */

import React, { Component } from 'react';
import { connect } from 'react-redux';
import {} from '../../../js/actions/index';
require('./_.sass');
class HomeContainer extends Component {
	constructor(props){
		super(props);
		this.state = {};
	}
	render() {
		return (
			/*Our return function MUST only return one element*/
			<div>
				<h1>HELLO WORLD</h1>
				{/*Once we have mapStateToProps properly setup,
				 we can just call this.props.[our data path]*/}
				<h2>This is {this.props.config.appName}</h2>
			</div>
		);
	}
}

/* mapStateToProps() is a super easy function that people
* make over complicated.  It just takes what is in our store
* and passes it to our container through this.props "Our containers properties"*/
function mapStateToProps(state) {
	return {
		config: state.config
	};
}

/* When we export just a component as it's name it is just a component
* ex: export default HomeContainer;
* When we add in connect, and connect our component to the mapStateToProps function,
* this is what gives our component the steroids to become the super hero container.
* */
export default connect(mapStateToProps, {})(HomeContainer);
