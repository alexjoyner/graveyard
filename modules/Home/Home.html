<div>
    <!-- Main Header -->
    <div id="">
        <h2 class="text-center">{{MainHeading}}</h2>
        <h5 class="text-center">{{quickInstructions}}</h5>
    </div>
    <!-- Tab Bar: Tab to Pros/Cons -->
    <ul class="nav nav-tabs">
        <li role="presentation" class="active"><a href="#/Home/All">All</a></li>
    </ul>
	<div ui-view></div>
</div>