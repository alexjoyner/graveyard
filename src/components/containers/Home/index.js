import React, { Component } from 'react';
import { connect } from 'react-redux';
//import HomeContainerTpl from './template';
import QuickLinks from '../../../components/panes/quick-links/index';
import {fetchGeneralFeedPosts, fetchTopicFeedPosts} from '../../../js/actions/index';
import Post from '../../../components/post/index';
require('./_.sass');
class HomeContainer extends Component {
	constructor(props){
		super(props);

		this.state = {headerText: ''};
	}
	componentWillReceiveProps(nextProps) {
		let params = nextProps.params;
		this.getPosts(params);
	}
	componentWillMount() {
		let params = this.props.params;
		this.getPosts(params);

	}
	getPosts(params){
		/* Check first for  */
		if (params.topic_name) {
			this.props.fetchTopicFeedPosts(params.topic_id);
			this.setState({headerText: 'Top posts in ' + params.topic_name})
			return;
		}
		if (params.feed_name === 'top') {
			this.setState({headerText: 'All time top posts'});
			this.props.fetchGeneralFeedPosts(params.feed_name);
			return;
		}
		this.setState({headerText: 'Hot questions on MetaTruth right now'});
		this.props.fetchGeneralFeedPosts();
	}
	renderPosts(posts){
		return posts.map(post => {
			return (
				<div key={post._id}>
					<Post post={post} />
				</div>
			)
		})
	}
	render() {
		return (
			<div id="Home-Container" className="row">
				<div className="col col-lg-3 hidden-lg-down">
					<QuickLinks/>
				</div>
				<div className="col col-lg-6 col-xs-12">
					<h5 id="Home-MainHeader">
						{this.state.headerText}
					</h5>
					<hr/>
					{this.renderPosts(this.props.posts)}
				</div>
				<div className="col col-lg-3 hidden-lg-down">

				</div>
			</div>
		);
	}
}

function mapStateToProps(state) {
	return { posts: state.posts.all };
}

// this construct gives us access to this.props.fetchPosts
// function mapDispatchToProps(dispatch) {
//   return bindActionCreators({ fetchPosts}, dispatch);
// }
//
// export default connect(null, mapDispatchToProps)(PostsIndex);
//
// we can shortcut the above boilerplate with:
// export default connect(null, {fetchPosts: fetchPosts})(PostsIndex);
//
// and further simplifying with ES6 syntax:
//
export default connect(mapStateToProps, { fetchGeneralFeedPosts, fetchTopicFeedPosts })(HomeContainer);