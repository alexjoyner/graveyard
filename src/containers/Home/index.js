import React, { Component } from 'react';
import { connect } from 'react-redux';
//import HomeContainerTpl from './template';
import QuickLinks from '../../components/panes/quick-links/index';
import fetchPosts from '../../js/actions/index';
import Post from '../../components/post/index';
require('./_.sass');
class HomeContainer extends Component {
	componentWillReceiveProps(nextProps) {
		let params = nextProps.params;
		/* Check first for  */
		if (params.feed_name) {
			this.props.fetchPosts(params.feed_name);
			return;
		}
		this.props.fetchPosts();
	}
	componentWillMount() {
		let params = this.props.params;
		/* Check first for  */
		if (params.feed_name) {
			this.props.fetchPosts(params.feed_name);
			return;
		}
		this.props.fetchPosts();
	}
	renderPosts(posts){
		console.log('Post: 0 ', posts[0]);
		return posts.map(post => {
			return (
				<div key={post._id}>
					<Post post={post}></Post>
				</div>
			)
		})
	}
	render() {
		return (
			<div className="row">
				<div className="col col-lg-2 hidden-lg-down">
					<div className="">
						<QuickLinks/>
					</div>
				</div>
				<div className="col col-lg-8 col-xs-12">
					{this.renderPosts(this.props.posts)}
				</div>
				<div className="col col-lg-2 hidden-lg-down">

				</div>
			</div>
		);
	}
}

function mapStateToProps(state) {
	return { posts: state.posts.all };
}

// this construct gives us access to this.props.fetchPosts
// function mapDispatchToProps(dispatch) {
//   return bindActionCreators({ fetchPosts}, dispatch);
// }
//
// export default connect(null, mapDispatchToProps)(PostsIndex);
//
// we can shortcut the above boilerplate with:
// export default connect(null, {fetchPosts: fetchPosts})(PostsIndex);
//
// and further simplifying with ES6 syntax:
//
export default connect(mapStateToProps, { fetchPosts })(HomeContainer);