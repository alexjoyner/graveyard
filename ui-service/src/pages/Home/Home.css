body{
  background-color: #4e4e4e;
}
.ees-heading{
  display: inline-block;
  background-color: #2d3e50;
  .ees-logo{
    font-size: 1.2em;
    color: white;
  }
}