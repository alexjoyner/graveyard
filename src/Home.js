import React, { Component } from 'react';
import logo from './react.svg';
import './Home.css';

class Home extends Component {
  render() {
    return (
      <div className="Home">
        <div className="Home-header">
          <img src={logo} className="Home-logo" alt="logo" />
          <h2 className="Greeting">Welcome to TEST JS</h2>
        </div>
      </div>
    );
  }
}

export default Home;
