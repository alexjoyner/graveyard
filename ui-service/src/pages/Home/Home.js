import React, {Component} from 'react';
import './Home.css';
import 'purecss';
import {GraphBlock} from "../../features/graph-block/index";
import Menu from "../../features/menu/index";
import Header from "../../features/header/index";
import {DragDropContext} from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';
import Dash from "../../features/Dash/index";

export class Home extends Component {
    constructor(){
        super();
        this.state = {
            graphs: []
        }
    }
    handleDashDrop = (data) => {
        let { graphs } = this.state;
        graphs.push(data);
        this.setState({graphs});
    };
    render() {
        return (
            <div>
                <Header/>
                <Menu/>
                <Dash onDrop={this.handleDashDrop}>
                    {/*{this.state.graphs.map((data, i) => {*/}
                        {/*return <GraphBlock key={i} BlockID={'Block' + i} data={data}/>*/}
                    {/*})}*/}
                </Dash>
            </div>
        );
    }
}
export default DragDropContext(HTML5Backend)(Home);
