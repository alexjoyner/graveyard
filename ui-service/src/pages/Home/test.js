import React from 'react';
import {shallow} from 'enzyme';
import {Home} from './Home';

describe('layout feature', () => {
    let component;
    let mockHandleDashDrop = jest.fn();
    beforeEach(() => {
        let props = {
            handleDashDrop: mockHandleDashDrop,
            graphs: []
        };
        component = shallow(<Home {...props}/>);
    });
    it('renders without exploding', () => {
        expect(component).toMatchSnapshot();
    });
    it('has a Header', () => {
        expect(component.find('Header').length).toEqual(1);
    });
    it('has a Menu', () => {
        expect(component.find('Menu').length).toEqual(1);
    });
    it('contains a Dash', () => {
        expect(component.find('DropTarget(Dash)').length).toEqual(1);
    });
    it('has a Graph Block', () => {
        expect(component.find('GraphBlock').length).toEqual(0);
    });
    it('should be able to call drop event', () => {
        let DashComponent = component.find('DropTarget(Dash)');
        DashComponent.props().onDrop({type: 'test'});
        expect(component.state().graphs).toEqual([{type: 'test'}]);
    });
    it('should add a Graph Block a new sensor is dropped onto the dash', () => {
        let DashComponent = component.find('DropTarget(Dash)');
        DashComponent.props().onDrop({type: 'test'});
        component.update();
        expect(component.find('GraphBlock').length).toEqual(1);
    });
});