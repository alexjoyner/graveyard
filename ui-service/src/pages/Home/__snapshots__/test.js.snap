// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout feature renders without exploding 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    graphs={Array []}
    handleDashDrop={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <Menu />,
        <DropTarget(Dash)
          onDrop={[Function]}
>
          
</DropTarget(Dash)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [],
          "onDrop": [Function],
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <Menu />,
          <DropTarget(Dash)
            onDrop={[Function]}
>
            
</DropTarget(Dash)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [],
            "onDrop": [Function],
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
